django/contrib/gis/gdal/raster/source.py

```diff
@@ -425,6 +425,21 @@ def warp(self, ds_input, resampling='NearestNeighbour', max_error=0.0):
 
         return target
 
+    def clone(self):
+        "Return a clone of this GDALRaster object"
+        return GDALRaster(
+            capi.copy_ds(
+                self.driver._ptr,
+                force_bytes(self.name),
+                self._ptr,
+                c_int(),
+                c_char_p(),
+                c_void_p(),
+                c_void_p()
+            ),
+            write=self._write
+        )
```

Comment:
Is there anything wrong with `return type(self)(capi.copy_ds(self.ptr))`? this should be the most efficient way.